kv/txn_coord_sender.go

```diff
@@ -758,28 +756,26 @@ func (tc *TxnCoordSender) heartbeat(id string) {
 					Reply: reply,
 				}
 
-				task := tc.stopper.StartTask()
-				if !task.Ok() {
+				if !tc.stopper.RunTask(func() {
```

Comment:
these return values should really go away, they just encourage strange control flow like the presence of "continue" below. Does that continue need to be there? will the value of `proceed` ever be false if this block doesn't run?